pytorch_lightning/core/datamodule.py

```diff
@@ -373,10 +373,13 @@ def dataloader(ds, shuffle=False):
 
         def train_dataloader():
             if isinstance(train_dataset, Mapping):
-                return {key: dataloader(ds, shuffle=True) for key, ds in train_dataset.items()}
+                return {
```

Comment:
nit: can you reuse this var?

```
should_shuffle = not isinstance(ds, IterableDataset)
```

Reply:
Since `should_shuffle` has to be checked for every instance of `Dataset`, I think there are two options for code reuse:

**Option 1: Declaring a `should_shuffle` function.**
![option1](https://user-images.githubusercontent.com/9260067/115252684-a7453980-a166-11eb-80f3-a943be7a7604.png)

**Option 2: Declaring a `should_shuffle` function.**
![option2](https://user-images.githubusercontent.com/9260067/115253549-73b6df00-a167-11eb-8358-c9d68eb94146.png)

Option 2 seems to be a major change from existing code behavior (introduces ways to override `shuffle`).

Option 1 seems to be personal preference, so right now I opted for no change. Any suggestions would be wonderful!

Reply:
I prefer the variable than the function in this case

Reply:
Should I commit with option 2? Or leave as-is?

Reply:
Made the change for you :)